x-pack/plugins/fleet/public/components/agent_enrollment_flyout/hooks.tsx

```diff
@@ -64,20 +72,71 @@ export function useIsK8sPolicy(agentPolicy?: AgentPolicy) {
           : 'IS_NOT_KUBERNETES'
       );
     }
+
+    async function checkifCSP() {
+      if (!agentPolicy) {
+        setIsCSP({ status: 'IS_LOADING', cloudformationUrl: undefined });
+        return;
+      }
+
+      const cspPackageObj = agentPolicy?.package_policies?.map((pkgPolicy) =>
+        getCSPPackageType(pkgPolicy)
+      );
+
+      const cloudFormationTemplateUrl = getCloudFormationTemplateUrlFromPackagePolicy(agentPolicy);
+
+      if (cspPackageObj?.includes('IS_CSP_KSPM')) {
+        setIsCSP({ status: 'IS_CSP_KSPM', cloudformationUrl: cloudFormationTemplateUrl });
+      } else if (cspPackageObj?.includes('IS_CSP_CSPM')) {
+        setIsCSP({ status: 'IS_CSP_CSPM', cloudformationUrl: cloudFormationTemplateUrl });
+      } else if (cspPackageObj?.includes('IS_CSP_CNVM')) {
+        setIsCSP({ status: 'IS_CSP_CNVM', cloudformationUrl: cloudFormationTemplateUrl });
+      } else {
+        setIsCSP({ status: 'IS_NOT_CSP', cloudformationUrl: cloudFormationTemplateUrl });
+      }
+    }
+
     checkifK8s();
+    checkifCSP();
   }, [agentPolicy]);
 
-  return { isK8s };
+  return { isK8s, isCSP };
 }
 
 const isK8sPackage = (pkg: PackagePolicy) => {
   const name = pkg.package?.name as string;
-  if (name === FLEET_CLOUD_SECURITY_POSTURE_PACKAGE) {
-    return pkg.inputs.some(
-      (input) =>
-        input.enabled && input.policy_template === FLEET_CLOUD_SECURITY_POSTURE_KSPM_POLICY_TEMPLATE
-    );
-  }
 
   return K8S_PACKAGES.has(name);
 };
+
+const getCSPPackageType = (pkg: PackagePolicy) => {
+  const name = pkg.package?.name as string;
+  if (name === FLEET_CLOUD_SECURITY_POSTURE_PACKAGE) {
+    if (
+      pkg.inputs.some(
+        (input) =>
+          input.enabled &&
+          input.policy_template === FLEET_CLOUD_SECURITY_POSTURE_KSPM_POLICY_TEMPLATE
+      )
+    ) {
+      return 'IS_CSP_KSPM';
+    } else if (
+      pkg.inputs.some(
+        (input) =>
+          input.enabled &&
+          input.policy_template === FLEET_CLOUD_SECURITY_POSTURE_CSPM_POLICY_TEMPLATE
+      )
+    ) {
+      return 'IS_CSP_CSPM';
+    } else if (
+      pkg.inputs.some(
+        (input) =>
+          input.enabled &&
+          input.policy_template === FLEET_CLOUD_SECURITY_POSTURE_CNVM_POLICY_TEMPLATE
+      )
+    ) {
+      return 'IS_CSP_CNVM';
+    }
+  }
+  return 'IS_NOT_CSP';
+};
```

Comment:
Following my previous comments, we then would be able to refactor this function  to something like that:

```js
const getCloudSecurityIntegrationTypeFromPackagePolicy = (packagePolicy: PackagePolicy) :  CloudSecurityIntegrationType | undefined {
	const name = packagePolicy.package?.name as string;
	if (name === FLEET_CLOUD_SECURITY_POSTURE_PACKAGE) {
		return package.inputs.find((input) => input.enabled)?.policy_template;
	}
	return undefined;
}
```